gq: fix build with autotools

barracuda156 opened the pull request:
#### Description

As suggested by @jmroot in: https://trac.macports.org/ticket/59054#comment:5

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

barracuda156 commented:
Well, now it fails:
```
LC_ALL=C ../intltool-merge -x -u -c ../po/.intltool-merge-cache ../po gq-ldif.xml.in gq-ldif.xml
  LC_ALL=C ../intltool-merge -d -u -c ../po/.intltool-merge-cache ../po gq.desktop.in gq.desktop
  /bin/sh: ../intltool-merge: -w: bad interpreter: No such file or directory
  make[2]: *** [gq-ldif.xml] Error 126
```

jmroot commented:
Oh, gq actually ships its own copies of the intltool executables, making this the first build system I've seen that actually uses the value of `INTLTOOL_PERL` after setting it. So ideally it would run the versions provided by the intltool port instead (assuming they are compatible). Maybe autoreconf with `-fvi` would help?

If that's not possible or too difficult, then gq would have to depend on intltool's dependencies so that it can guarantee the availability of the necessary perl modules for the perl version it is currently using.

barracuda156 commented:
@jmroot I can try using autotools. What is the criterion of success? FWIU it was building fine before, at least both CI bots and locally on my machines.

jmroot commented:
The bug makes it fail to build when whatever `perl` configure happens to find doesn't have the XML::Parser module. It works accidentally if you have the perl5 port installed with the variant matching the perl version that intltool currently uses.

jmroot commented:
This isn't theoretical by the way, you can see many, many tickets involving this failure mode in various ports by searching Trac for `XML::Parser`: https://trac.macports.org/search?q=XML%3A%3AParser&noquickjump=1&ticket=on

barracuda156 commented:
@jmroot I tried autotools, but it fails with:
```
configure.in:439: You should run autoupdate.
configure.in:5: error: possibly undefined macro: AM_ACLOCAL_INCLUDE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.in:67: error: possibly undefined macro: AS_PROG_OBJC
autoreconf: error: /opt/local/bin/autoconf failed with exit status: 1
```

To be explicit, I dropped intltool-perl patch and added this:
```
use_autoreconf  yes

depends_build-append \
                port:autoconf \
                port:automake \
                port:intltool \
                port:libtool \
                port:pkgconfig
```

ryandesign commented:
> ```
> configure.in:5: error: possibly undefined macro: AM_ACLOCAL_INCLUDE
> ```

[A Google search of that error message suggests](https://wiki.mageia.org/en/Packaging_problems_and_solutions#error:_possibly_undefined_macro:_AM_ACLOCAL_INCLUDE) replacing `AM_ACLOCAL_INCLUDE` with `ACLOCAL_AMFLAGS`.

barracuda156 commented:
@jmroot @ryandesign Finally it worked. Needed one more fix to `configure.in`.

jmroot commented:
You need to autoreconf so the changes to configure.in and Makefile.am are actually used.

barracuda156 commented:
> You need to autoreconf so the changes to configure.in and Makefile.am are actually used.

Sorry, my bad, autoreconf block somehow was missed (I have it locally, it vanished in rebasing). Gonna update now.

jmroot commented on net/gq/Portfile:
```diff
@@ -21,7 +21,13 @@ checksums       rmd160  4e8bf1700f4d7aacd906559e01a11654f6ce0f90 \
+                port:autoconf \
+                port:automake \
+                port:intltool \
+                port:libtool \
```
The autotools deps are added automatically by `use_autoreconf  yes`.

barracuda156 replied:
Do we still need `intltool` there?

jmroot replied:
Yes.

barracuda156 replied:
Good now? Check passed.

jmroot reviewed (approved):
Thanks!
